src/Diagnostics/diagnostic_fields.jl

```diff
@@ -0,0 +1,294 @@
+using CUDAdrv
+using CUDAnative
+using CUDAapi
+using CuArrays
+using StaticArrays
+using KernelAbstractions
+using CLIMA.Mesh.Geometry
+
+const _ξ1x1, _ξ2x1, _ξ3x1 = Grids._ξ1x1, Grids._ξ2x1, Grids._ξ3x1
+const _ξ1x2, _ξ2x2, _ξ3x2 = Grids._ξ1x2, Grids._ξ2x2, Grids._ξ3x2
+const _ξ1x3, _ξ2x3, _ξ3x3 = Grids._ξ1x3, Grids._ξ2x3, Grids._ξ3x3
+const _ρ, _ρu, _ρv, _ρw = 1, 2, 3, 4
```

Comment:
Shouldn't we use `vars_state` instead of hard-coded indexes?

Reply:
This is a first rough draft for backup purposes. Removing hard-code indexes is on the TODO list.

Reply:
see if you can use `varsindex` for this?

Reply:
https://github.com/climate-machine/CLIMA/blob/0ca53bec15528e1b62814307611212907481513e/test/Utilities/VariableTemplates/varsindex.jl

Reply:
https://github.com/climate-machine/CLIMA/blob/9015194811b1d3edd75bb2dc0601ac7e70912e4c/examples/Microphysics/ex_1_saturation_adjustment.jl#L242-L247